File: examples/dart/language_gauntlet/lib/src/domain/envelope.dart
enum Kind { run, eval }

class Envelope {
  final Kind kind;
  final String cmd;
  final String user;
  final int length;
  final List<String> extras;

  Envelope({
    required this.kind,
    required this.cmd,
    required this.user,
    required this.length,
    required this.extras,
  });

  Envelope copyWith({Kind? kind, String? cmd, String? user, int? length, List<String>? extras}) =>
      Envelope(
        kind: kind ?? this.kind,
        cmd: cmd ?? this.cmd,
        user: user ?? this.user,
        length: length ?? this.length,
        extras: extras ?? this.extras,
      );
}

File: examples/csharp/language_gauntlet/Domain/Envelope.cs
namespace LanguageGauntlet.Domain;

using System.Collections.Generic;

public enum Kind { Run, Eval }

public record Envelope(Kind Kind, string Cmd, string User, int Length, List<string> Extras);

public static class EnvelopeFactory
{
    public static Envelope Create(string cmd, string user) =>
        new(Kind.Run, Cmd: $"{cmd}", User: user, Length: cmd.Length, Extras: new List<string> { cmd });
}
